cdap-ui/app/features/hydratorplusplus/controllers/create/toppanel-ctrl.js

```diff
@@ -157,9 +157,124 @@ class HydratorPlusPlusTopPanelCtrl{
   showSettings() {
     this.state.viewSettings = !this.state.viewSettings;
   }
-}
 
-HydratorPlusPlusTopPanelCtrl.$inject = ['$stateParams', 'HydratorPlusPlusConfigStore', 'HydratorPlusPlusConfigActions', '$uibModal', 'HydratorPlusPlusConsoleActions', 'DAGPlusPlusNodesActionsFactory', 'GLOBALS', 'myHelpers', 'HydratorPlusPlusConsoleStore', 'myPipelineExportModalService', '$timeout', '$scope'];
+  // PREVIEW
+  startTimer() {
+    this.previewTimerInterval = this.$interval(() => {
+      let duration = (new Date() - this.previewStartTime) / 1000;
```

Comment:
On a sidenote it would be nice if this is made into a separate component as we will be using this in the published pipeline view too (duration of run).